internal/datanode/compaction/executor.go

```diff
@@ -50,8 +52,10 @@ type executor struct {
 	completedCompactor *typeutil.ConcurrentMap[int64, Compactor]                    // planID to compactor
 	completed          *typeutil.ConcurrentMap[int64, *datapb.CompactionPlanResult] // planID to CompactionPlanResult
 	taskCh             chan Compactor
-	taskSem            *semaphore.Weighted
+	taskSem            *semaphore.Weighted             // todo remove this, unify with slot logic
 	dropped            *typeutil.ConcurrentSet[string] // vchannel dropped
+	usingSlots         *atomic.Int64
```

Comment:
Do we need this?

Reply:
Actually it is a double check. Executor can refuse new task even datacoord works unexpectedly.